lib/apis/v3/bgppeer.go

```diff
@@ -44,8 +44,10 @@ type BGPPeerSpec struct {
 	// If this is not set, this represents a global peer, i.e. a peer that peers with
 	// every node in the deployment.
 	Node string `json:"node,omitempty" validate:"omitempty,name"`
-	// The IP address of the peer.
-	PeerIP string `json:"peerIP" validate:"omitempty,ip"`
+	// The IP address of the peer followed by an optional port number to peer with.
+	// If port number is given, format should be `[<IPv6>]:port` or `<IPv4>:<port>` for IPv4.
+	// Default port number is 179.
+	PeerIP string `json:"peerIP" validate:"omitempty"`
```

Comment:
It might be nice to define a new validator type for the new "IP:port" type so it could be re-used elsewhere if / when needed.

I think this PR just includes this in the validator code for the BGPPeerSpec type?

Reply:
right, it was in validator for BGPPeerSpec, I've refactored it now.